### PR TITLE
ml - use BigDecimal instead of String for amounts

### PR DESCRIPTION
**Pull request by magneland**
Note: Rails as_json converts String, so don’t use that

Also change assert to assert_equal
Note: assert ignores the second arg :-(

### CODE REVIEW

File: lib/rentlinx/validators/amount_validator.rb [hunk 1]
```diff
@@ -6,8 +6,7 @@ class AmountValidator < BaseValidator
     private
 
     def validate
-      @value.sub!(/^0+/, '') unless @value.nil?
-      return if @value.to_s =~ /[1-9]\d*(\.\d{1,2})?/
+      return if !@value.nil? && @value.is_a?(Numeric) && @value >= 0
```
Comment (XanderStrike) [inline lib/rentlinx/validators/amount_validator.rb:9]: Should `@value.is_a?(Numeric)` not be `@value.is_a?(BigDecimal)`?

Comment (magneland) [inline lib/rentlinx/validators/amount_validator.rb:9]: Yeah I guess. I was trying to be more general, but it's probably not worth it.

Comment (magneland) [inline lib/rentlinx/validators/amount_validator.rb:9]: Amended commit.

Comment (magneland) [inline lib/rentlinx/validators/amount_validator.rb:9]: Amended commit again :) to satisfy `VCR`

**Pull request merged**